code/modules/defenses/sentry_computer.dm

```diff
@@ -0,0 +1,439 @@
+#define DEFAULT_MAP_SIZE 15
+
+/obj/item/device/sentry_computer
+	name = "sentry computer"
+	desc = "A laptop loaded with sentry control software."
+	icon = 'icons/obj/structures/props/sentrycomp.dmi'
+	icon_state = "sentrycomp_cl"
+	var/setup = FALSE
+	var/open = FALSE
+	var/on = FALSE
+	var/obj/structure/machinery/defenses/sentry/paired_sentry = list()
+	var/obj/structure/machinery/defenses/sentry/current
+	var/cell_type = /obj/item/cell/high
+	var/obj/item/cell/cell
+	var/power_consumption = 1
+	w_class = SIZE_SMALL
+	var/state = 0
+	var/screen_state = 0
+	var/list/registered_tools = list()
+	var/list/faction_group
+	var/silent = FALSE
+	var/can_identify_target = FALSE
+
+	/// The turf where the camera was last updated.
+	var/turf/last_camera_turf
+
+	var/obj/item/device/radio/marine/transceiver = new /obj/item/device/radio/marine
+	var/mob/living/voice = new /mob/living/silicon
+
+	// Stuff needed to render the map
+	var/map_name
+	var/atom/movable/screen/map_view/cam_screen
+	var/atom/movable/screen/background/cam_background
+
+	/// All turfs within range of the currently active camera
+	var/list/range_turfs = list()
+
+/obj/item/device/sentry_computer/Initialize(mapload)
+	. = ..()
+	if(cell_type)
+		cell = new cell_type()
+		cell.charge = cell.maxcharge
+	// set up cameras
+	map_name = "sentry_computer_[REF(src)]_map"
+	cam_screen = new
+	cam_screen.name = "screen"
+	cam_screen.assigned_map = map_name
+	cam_screen.del_on_map_removal = FALSE
+	cam_screen.screen_loc = "[map_name]:1,1"
+	cam_background = new
+	cam_background.assigned_map = map_name
+	cam_background.del_on_map_removal = FALSE
+
+	faction_group = FACTION_LIST_MARINE
+	transceiver.forceMove(src)
+	transceiver.set_frequency(SENTRY_FREQ)
+	transceiver.config(list(RADIO_CHANNEL_SENTRY=1))
+	transceiver.subspace_transmission = TRUE
+	voice.name = "[name]:[serial_number]"
+	voice.forceMove(src)
+
+/obj/item/device/sentry_computer/Destroy()
+	. = ..()
+	QDEL_NULL(cell)
+	qdel(cam_background)
+	qdel(cam_screen)
+	qdel(transceiver)
+	qdel(voice)
+
+/obj/item/device/sentry_computer/Move(NewLoc, direct)
+	..()
+	if(setup || open || on)
+		teardown()
+
+
+/obj/item/device/sentry_computer/proc/has_los(var/atom/watcher, var/atom/target)
+	var/list/turf/path = getline2(watcher, target, include_from_atom = FALSE)
+	for(var/turf/point in path)
+		if(point.opacity)
+			return FALSE
```

Comment:
```suggestion
		if(!point.opacity)
			continue
		return FALSE
```

Reply:
I'm not sure how this is an improvement, we're in a loop and if the check fails it will continue anyway